src/terminal/input/terminalInput.hpp

```diff
@@ -48,7 +70,62 @@ namespace Microsoft::Console::VirtualTerminal
         bool _cursorApplicationMode = false;
 
         void _SendNullInputSequence(const DWORD dwControlKeyState) const;
-        void _SendInputSequence(const std::wstring_view sequence) const;
+        void _SendInputSequence(const std::wstring_view sequence) const noexcept;
         void _SendEscapedInputSequence(const wchar_t wch) const;
+
+#pragma region MouseInputState Management
+        // These methods are defined in mouseInputState.cpp
+        enum class ExtendedMode : unsigned int
+        {
+            None,
+            Utf8,
+            Sgr,
+            Urxvt
+        };
+
+        enum class TrackingMode : unsigned int
+        {
+            None,
+            Default,
+            ButtonEvent,
+            AnyEvent
+        };
+
+        struct MouseInputState

```

Comment:
I mean, I probably would have just made the `MouseInput` class a child member of the `TerminalInput` class instead of merging the two classes, but ¯\\\_(ツ)\_/¯

Reply:
haha. I tried that a few months ago but it didn't work and I got frustrated. I figured this time, I'd start with this then separate it out again as before... but now it works...and I can just hand a TerminalInput around instead of MouseInput + TerminalInput. So, meh. I relate haha